### PR TITLE
Setup GitHub Actions

### PR DESCRIPTION
**Pull request by bmoffatt**

### CODE REVIEW

**Comment by bmoffatt** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: aws-lambda-go-github-ci
* Commit ID: 4c06f5c0d3cfdafda06bd719a432830d500f9b70
* Result: SUCCEEDED
* [Build Logs](https://yxbry9e1t8.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=53ef733d-596c-44e0-8c52-5988f6994360%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by bmoffatt** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: aws-lambda-go-github-ci
* Commit ID: a6647f8d7997b9e0e5cd97cd50118742cb8732bd
* Result: SUCCEEDED
* [Build Logs](https://yxbry9e1t8.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ef139f03-460b-4166-b11e-16d2cbf31994%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: .github/workflows/build-lambda-zip.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: go get build-lambda-zip 
+on:
+  push:
+  schedule:
+    - cron: "7 7 * * *"
```
Comment (harrisonhjones) [inline .github/workflows/build-lambda-zip.yml]: preface: I'm new to GitHub Actions.

Perhaps for my own learning: what's the value in regularly running this action (over on push)?

File: .github/workflows/build-lambda-zip.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: go get build-lambda-zip
```
Comment (harrisonhjones) [inline .github/workflows/build-lambda-zip.yml]: Looks like you can specify a [description](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/metadata-syntax-for-github-actions#description). Imo it's a bit unclear *why* this action exists and perhaps a short explanation in the description would be beneficial?

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: tests
+on: 
+  push:
+  pull_request:
+  schedule:
+    - cron: "7 7 * * *"
```
Comment (harrisonhjones) [inline .github/workflows/tests.yml]: Similar comment here: What's the value of running our tests regularly? We are using Go Modules which pins our dependencies (afaik) and we have our Go version pinned so unless Go introduces a breaking change why would we need to run our tests on a schedule?

**Pull request merged**